scrubadub/detectors/date_of_birth.py

```diff
@@ -48,14 +55,54 @@ def iter_filth(self, text: str, document_name: Optional[str] = None):
                     if datetime.now().year - identified_date[1].year >= self.min_age_years and \
                             any(trigger_word in ' '.join(lines[count - self.context_before:count + self.context_after]
                                                          ).lower() for trigger_word in self.trigger_words):
-                        # date_picker doesn't return the startend, so using re.finditer for the start & end locations
+                        # date_picker doesn't return the start end, so using re.finditer for the start & end locations
                         found_dates = re.finditer(re.escape(r"{0}".format(identified_date[0])), text)
-                        # Iterate over each found string matching this regex and yield some filth
+
                         for instance in found_dates:
                             yield self.filth_cls(
                                 beg=instance.start(),
                                 end=instance.end(),
                                 text=instance.group(),
                                 detector_name=self.name,
                                 document_name=document_name,
+                                locale=self.locale,
                             )
+
+    def nuke_dob(self, text: str, document_name: Optional[str] = None):
```

Comment:
Does removing the below lines:
```
                            any(trigger_word in ' '.join(lines[count - self.context_before:count + self.context_after]
                                                         ).lower() for trigger_word in self.trigger_words):
```
from context_dob give you nuke_dob? If so maybe the code can be refactored to reduce duplication?

Maybe this is a good opportunity to reduce the complexity of the if statement on L55-L57 to make it more flexible?